File: з1.py
def f1(x):
    result = 1
    for j in range(len(x)):
        result *= x[j]
    return result

def f2(x):
    result = 1
    for j in range(len(x)):
        if x[j] % 2 == 0:
            result *= x[j]
    return result

n = int(input('Введите размер массива: '))
a = [0] * n

print('Введите элементы массива:')
for i in range(n):
    a[i] = int(input())

print('Ваш массив:')
for i in range(n):
    print(a[i], end=' ')
print()

print('Произведение всех элементов:', f1(a))
print('Произведение четных элементов:', f2(a))

File: з1.pas
program one;
var  a: array of integer; // a - глобальная переменная
  n, i: integer; // n,i - глобальные переменные 
function f1(x: array of integer): integer; // x - формальный параметр(передача по ссылке)
var j:integer; // локальная переменная
begin
  Result := 1; // локальная переменная
  for j := 0 to Length(x) - 1 do
    Result := Result * x[j];
end;
function f2(x: array of integer): integer; // x - формальный параметр(передача по ссылке)

var j:integer; // локальная переменная  
begin
  Result := 1; // локальная переменная
  for  j := 0 to Length(x) - 1 do 
  begin
    if x[j] mod 2 = 0 then
      Result := Result * x[j];
  end;
end;
begin
  write('Введите размер массива: ');
  readln(n);
  SetLength(a, n);
  writeln('Введите элементы массива:');
  for i := 1 to n do
  begin
    readln(a[i-1]);
  end;
  writeln('Ваш массив:');
  for i := 0 to n - 1 do
    write(a[i], ' ');
  writeln;
  writeln('Произведение всех элементов: ', f1(a)); //a - фактический параметр
  writeln('Произведение четных элементов: ', f2(a)); //a - фактический параметр
end.